omod/src/main/java/org/openmrs/module/laboratoryapp/fragment/controller/ResultFragmentController.java

```diff
@@ -73,9 +72,53 @@ public SimpleObject saveResult(
 		test.setEncounter(encounter);
 		test = ls.saveLaboratoryTest(test);
 		ls.completeTest(test);
-		
+
+		this.sendPatientToQueue(encounter);
+
 		return SimpleObject.create("status", "success", "message", "Saved!");
 	}
+
+	private void sendPatientToQueue(Encounter encounter)
+	{
+		//send patient back to the OPD Queue
+		Patient patient = encounter.getPatient();
+		PatientQueueService queueService = Context.getService(PatientQueueService.class);
+		Concept referralConcept = Context.getConceptService().getConcept(2548);
+		Encounter queueEncounter = queueService.getLastOPDEncounter(encounter.getPatient());
+		OpdPatientQueueLog patientQueueLog =queueService.getOpdPatientQueueLogByEncounter(queueEncounter);
+		Concept selectedOPDConcept = patientQueueLog.getOpdConcept();
+		String selectedCategory = patientQueueLog.getCategory();
+		String visitStatus = patientQueueLog.getVisitStatus();
+
+		OpdPatientQueue queue = queueService.getOpdPatientQueue(
+				patient.getPatientIdentifier().getIdentifier(), selectedOPDConcept.getConceptId());
+
+		if (queue == null) {
```

Comment:
remember the condition reads `if (queue == null)`; so only execute if patient is not in queue